docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md

```diff
@@ -0,0 +1,201 @@
+
+:::note
+If both feature toggles are enabled, `ENABLE_CONF_GENERATION_ON_DEPLOY` will prevail.
+:::
+
+## Migrating your projects
+
+Some essential manual adjustments are necessary to make sure your newly created and already existing projects work correctly with the Enhanced Project Workflow.
+
+### General adjustments
+
+The migration process for an existing Project is currently completely manual and impacts several entities, the process can be summarized with the following steps:
+
+ - Git repository structure clean-up
+ - Feature activation
+ - Setup default revision and few Project Settings
+
+#### Git repository structure clean-up
+
+With the new Workflow we have decided to change and unify the Git repository structure; prior to this change we are supporting two different folder structures based on whether the Project is using [`kustomize`](/console/project-configuration/kustomize-your-configurations/migrate-to-kustomize.md); now we are supporting a single folder structure that contains:
+
+- an `environments` directory, automatically generated by the Console, containing one directory for each environment, inside which all the necessary configurations are generated and stored after each push;
+- a `configuration` directory, with all your global customizations to be applied to all environments, and a `kustomization.yaml` manifest if you are using [`Kustomize`](/console/project-configuration/kustomize-your-configurations/manage-a-kustomize-project);
+- an `overlays` directory, containing one directory for each environment, inside which all your environment-specific patch files are stored, as well as a `kustomization.yaml` manifest if you are using [`Kustomize`](/console/project-configuration/kustomize-your-configurations/manage-a-kustomize-project).
+
+The final repository structure should be similar to the following: 
+
+```
+├── overlays
+│ ├── dev
+│ │ └─── <dev custom file>
+│ ├── production
+│ │ └─── <prod custom file>
+│ ├── other_env_id
+│ │ └─── <env_id custom file>
+│ └── ....
+│
+├── configuration
+│ └─── <global custom files>
+│
+├── environments
+│ ├── dev
+│ │ └── <k8s config files>
+│ ├── production
+│ │ └── <k8s config files>
+│ ├── other_env_id
+│ │ └── <k8s config files>
+│ └ .....
+```
+
+:::info
+All the contents of the `environments` directory are generated from scratch each time, any additional file that is not managed by the Console will be overwritten!
+:::
+
+:::caution
+If you are already working on a Project based on Kustomize, make sure that the `overlays/:env/kustomization.yml` file imports the proper resource within the `environments/:env/kustomization.yml` instead of the `configuration/kustomization.yml`. The Console, when generating the `environments/:env/kustomization.yml` will automatically import the `configuration/kustomization.yml` if present.
```

Comment:
Why not:

```suggestion
If you are working on a Project with Kustomize, ensure that the `overlays/:env/kustomization.yml` file imports the proper resource. 
To do this, replace the path `../../configuration` with `../../environments/<environment-id>` within the `environments/:env/kustomization.yml` instead of the `configuration/kustomization.yml`.
When generating the `environments/:env/kustomization.yml`, the Console will automatically import the `configuration/kustomization.yml` if present.
```